### PR TITLE
[ufc] update "get_assignment" args ordering to comply with ufc spec

### PR DESCRIPTION
**Pull request by schmit**
We forgot to update the ordering of arguments for the `get_assignment` functions -- fixed here

### CODE REVIEW

File: eppoclient/eppoclient_e2e_test.go [hunk 1]
```diff
@@ -47,20 +47,20 @@ func Test_e2e(t *testing.T) {
 				t.Run(subject.SubjectKey, func(t *testing.T) {
 					switch test.VariationType {
 					case booleanVariation:
-						value, _ := client.GetBoolAssignment(subject.SubjectKey, test.Flag, subject.SubjectAttributes, test.DefaultValue.(bool))
+						value, _ := client.GetBoolAssignment(test.Flag, subject.SubjectKey, subject.SubjectAttributes, test.DefaultValue.(bool))
 						assert.Equal(t, subject.Assignment, value)
 					case numericVariation:
-						value, _ := client.GetNumericAssignment(subject.SubjectKey, test.Flag, subject.SubjectAttributes, test.DefaultValue.(float64))
+						value, _ := client.GetNumericAssignment(test.Flag, ssubject.SubjectKey, ubject.SubjectAttributes, test.DefaultValue.(float64))
```
Comment (giorgiomartini0) [inline eppoclient/eppoclient_e2e_test.go:53]:
```suggestion
						value, _ := client.GetNumericAssignment(test.Flag, subject.SubjectKey, subject.SubjectAttributes, test.DefaultValue.(float64))
```

File: eppoclient/client.go [hunk 2]
```diff
@@ -36,8 +36,8 @@ func (ec *EppoClient) GetBoolAssignment(subjectKey string, flagKey string, subje
 	return result, err
 }
 
-func (ec *EppoClient) GetNumericAssignment(subjectKey string, flagKey string, subjectAttributes SubjectAttributes, defaultValue float64) (float64, error) {
-	variation, err := ec.getAssignment(subjectKey, flagKey, subjectAttributes, numericVariation)
+func (ec *EppoClient) GetNumericAssignment(flagKey string, subjectKey string, subjectAttributes SubjectAttributes, defaultValue float64) (float64, error) {
+	variation, err := ec.getAssignment(flagKey, subjectKey,subjectAttributes, numericVariation)
```
Comment (giorgiomartini0) [inline eppoclient/client.go:40]:
```suggestion
	variation, err := ec.getAssignment(flagKey, subjectKey, subjectAttributes, numericVariation)
```

File: eppoclient/client.go [hunk 3]
```diff
@@ -48,8 +48,8 @@ func (ec *EppoClient) GetNumericAssignment(subjectKey string, flagKey string, su
 	return result, err
 }
 
-func (ec *EppoClient) GetIntegerAssignment(subjectKey string, flagKey string, subjectAttributes SubjectAttributes, defaultValue int64) (int64, error) {
-	variation, err := ec.getAssignment(subjectKey, flagKey, subjectAttributes, integerVariation)
+func (ec *EppoClient) GetIntegerAssignment(flagKey string, subjectKey string, subjectAttributes SubjectAttributes, defaultValue int64) (int64, error) {
+	variation, err := ec.getAssignment(flagKey, subjectKey,subjectAttributes, integerVariation)
```
Comment (giorgiomartini0) [inline eppoclient/client.go:52]:
```suggestion
	variation, err := ec.getAssignment(flagKey, subjectKey, subjectAttributes, integerVariation)
```

File: eppoclient/client.go [hunk 4]
```diff
@@ -60,8 +60,8 @@ func (ec *EppoClient) GetIntegerAssignment(subjectKey string, flagKey string, su
 	return result, err
 }
 
-func (ec *EppoClient) GetStringAssignment(subjectKey string, flagKey string, subjectAttributes SubjectAttributes, defaultValue string) (string, error) {
-	variation, err := ec.getAssignment(subjectKey, flagKey, subjectAttributes, stringVariation)
+func (ec *EppoClient) GetStringAssignment(flagKey string, subjectKey string, subjectAttributes SubjectAttributes, defaultValue string) (string, error) {
+	variation, err := ec.getAssignment(flagKey, subjectKey,subjectAttributes, stringVariation)
```
Comment (giorgiomartini0) [inline eppoclient/client.go:64]:
```suggestion
	variation, err := ec.getAssignment(flagKey, subjectKey, subjectAttributes, stringVariation)
```

File: eppoclient/client.go [hunk 5]
```diff
@@ -72,15 +72,15 @@ func (ec *EppoClient) GetStringAssignment(subjectKey string, flagKey string, sub
 	return result, err
 }
 
-func (ec *EppoClient) GetJSONAssignment(subjectKey string, flagKey string, subjectAttributes SubjectAttributes, defaultValue interface{}) (interface{}, error) {
-	variation, err := ec.getAssignment(subjectKey, flagKey, subjectAttributes, jsonVariation)
+func (ec *EppoClient) GetJSONAssignment(flagKey string, subjectKey string, subjectAttributes SubjectAttributes, defaultValue interface{}) (interface{}, error) {
+	variation, err := ec.getAssignment(flagKey, subjectKey,subjectAttributes, jsonVariation)
```
Comment (giorgiomartini0) [inline eppoclient/client.go:76]:
```suggestion
	variation, err := ec.getAssignment(flagKey, subjectKey, subjectAttributes, jsonVariation)
```

**Review by giorgiomartini0 — APPROVED** [review]

**Review by rasendubi — APPROVED** [review]

**Comment by rasendubi** [comment]
Pushed a commit to update documentation and bump version.

File: README.md [hunk 6]
```diff
@@ -22,14 +22,14 @@ In your `go.mod`, add the SDK package as a dependency:
 
 ```
 require (
-    github.com/Eppo-exp/golang-sdk/v3
+    github.com/Eppo-exp/golang-sdk/v4
```
Comment (leoromanovsky) [inline README.md:25]: 📈

**Review by leoromanovsky — APPROVED** [review]

**Pull request merged**